.cicd/create-docker-from-binary.sh

```diff
@@ -0,0 +1,81 @@
+#!/bin/bash
+
+set -euo pipefail
+
+# buildkite-agent artifact download someversion.deb .
+
+buildkite-agent artifact download '*.deb' --step ':ubuntu: Ubuntu 18.04 - Package Builder' .
+echo ":done: download successfull"
+
+SANITIZED_BRANCH=$(echo "$BUILDKITE_BRANCH" | sed 's.^/..' | sed 's/[:/]/_/g')
+SANITIZED_TAG=$(echo "$BUILDKITE_TAG" | sed 's.^/..' | tr '/' '_')
+echo "$SANITIZED_BRANCH"
+echo "$SANITIZED_TAG"
+
+# do docker build
+echo ":docker::build: Building image..."
+DOCKERHUB_EOS_REGISTRY="docker.io/eosio/eos"
+
+if [[ ! -z "$SANITIZED_TAG" ]]; then
+    IMAGE_ECR_C="$EOSIO_REGISTRY:$BUILDKITE_COMMIT"
+    IMAGE_ECR_B="$EOSIO_REGISTRY:$SANITIZED_BRANCH"
+    IMAGE_DOCKER_C="$DOCKERHUB_EOS_REGISTRY:$BUILDKITE_COMMIT"
+    IMAGE_DOCKER_B="$DOCKERHUB_EOS_REGISTRY:$SANITIZED_BRANCH"
+    DOCKER_BUILD_COMMAND="docker build -t $IMAGE_ECR_C -t $IMAGE_DOCKER_C -t $IMAGE_ECR_B -t $IMAGE_DOCKER_B -f ./docker/dockerfile ."
+    echo "Building Image...."
+    echo "$ $DOCKER_BUILD_COMMAND"
+    eval $DOCKER_BUILD_COMMAND
+    echo "Build Done successfully!!"
+        if [[ ! -z "$BUILDKITE_TAG" && "$SANITIZED_BRANCH" != "$SANITIZED_TAG" ]]; then
+            IMAGE_ECR_T="$EOSIO_REGISTRY:$SANITIZED_TAG"
+            IMAGE_DOCKER_T="$DOCKERHUB_EOS_REGISTRY:$SANITIZED_TAG"
+            DOCKER_TAG_ECR="docker tag '$IMAGE_ECR_C' '$IMAGE_ECR_T"
+            DOCKER_TAG_HUB="docker tag '$IMAGE_DOCKER_C '$IMAGE_DOCKER_T"
+        fi    
+fi
```

Comment:
I think we could improve this process by running a docker build first, then tag and push afterwards. This could help simplify the number of variables being maintained (and probably greatly simplify lines 19 through 74.

The initial build could just use some generic tag name at first.
```
docker build -t eos -f ./docker/dockerfile .
```

Then we can create a list of our registries to push to and loop to tag and push. Something like:
```
EOSIO_REGISTRIES=("dockerhub" "$OTHER_REGISTRY")
for REG in ${EOSIO_REGISTRIES[@]}; do
  docker tag eos REG:$BUILDKITE_COMMIT
  docker tag eos REG:$BUILDKITE_BRANCH
  docker push REG:$BUILDKITE_COMMIT
  docker push REG:$BUILDKITE_BRANCH
  docker rmi REG:$BUILDKITE_COMMIT
  docker rmi REG:$BUILDKITE_BRANCH
  if [[ ! -z $BUILDKITE_TAG ]]; then
    docker tag eos REG:$BUILDKITE_TAG
    docker push REG:$BUILDKITE_TAG
    docker rmi REG:$BUILDKITE_TAG
  fi
done
docker rmi eos
```

Reply:
Good insights about the structure and the container steps, I did it following the patterns indicated by Zach and your recommendations about the loop, also I ran some test to validate that all is working as expected.